services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/validation/CreateHankeRequestValidator.kt

```diff
@@ -0,0 +1,62 @@
+package fi.hel.haitaton.hanke.validation
+
+import fi.hel.haitaton.hanke.HankeError
+import fi.hel.haitaton.hanke.MAXIMUM_HANKE_NIMI_LENGTH
+import fi.hel.haitaton.hanke.domain.CreateHankeRequest
+import fi.hel.haitaton.hanke.domain.HankePerustaja
+import fi.hel.haitaton.hanke.validation.Validators.notBlank
+import fi.hel.haitaton.hanke.validation.Validators.notLongerThan
+import fi.hel.haitaton.hanke.validation.Validators.validate
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import kotlin.reflect.KClass
+
+@Target(AnnotationTarget.FIELD, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.PROPERTY_GETTER)
+@Retention(AnnotationRetention.RUNTIME)
+@Constraint(validatedBy = [CreateHankeRequestValidator::class])
+@MustBeDocumented
+annotation class ValidCreateHankeRequest(
+    val message: String = "",
+    val groups: Array<KClass<*>> = [],
+    val payload: Array<KClass<out Payload>> = []
+)
+
+class CreateHankeRequestValidator :
+    ConstraintValidator<ValidCreateHankeRequest, CreateHankeRequest> {
+    override fun isValid(
+        request: CreateHankeRequest?,
+        context: ConstraintValidatorContext
+    ): Boolean {
+        if (request == null) {
```

Comment:
Is it a possible scenario that the request is null? It is non nullable in the controller endpoint.

Reply:
Or is this executed before that? In that case this should be ok to provide a more specific error.

Reply:
I couldn't find a scenario where the request would be null. When I tried with a broken or an empty body, I got an `org.springframework.http.converter.HttpMessageNotReadableException` exception before the validator was called. I made the request non-nullable, which made this method much simpler. Good catch.